Dockerfile.mustgather

```diff
@@ -0,0 +1,9 @@
+FROM registry.ci.openshift.org/ocp/4.10:must-gather AS builder
+WORKDIR /go/src/github.com/openshift/csi-driver-shared-resource
+COPY . .
+
+FROM registry.ci.openshift.org/ocp/4.10:cli
+COPY --from=builder /go/src/github.com/openshift/csi-driver-shared-resource/must-gather/* /usr/bin/
```

Comment:
Why do we need to run a multi-stage build? Isn't the base `cli` image sufficient, since this is just running bash?

Reply:
this is the pattern exhibited elsewhere in multiple repos.

it *could* be an oversight that has been copy/pasted, but my assumption was there were elements in the must-gather image that are needed so that `oc adm must-gather` seemlessly calls this when used in conjunction with other must gather image.  You can call `oc adm must-gather` where you specify both the base must-gather imagestream, plus a feature image, as well as calling it with just the feature image.

if we want to fully satisfy your curiosity here, sure, and can do some experiments where I don't use multi-stage builds, and see what works / what does not work.

Reply:
Before I try several permutations @adambkaplan do you have a precise Dockerfile you think should work?

Reply:
So I tried 4 permutations.  2 failed on the image build.  Two at least built as images, but failed on execution.

```
FROM registry.ci.openshift.org/ocp/4.10:cli
WORKDIR /go/src/github.com/openshift/csi-driver-shared-resource
COPY . .
RUN chmod +x /go/src/github.com/openshift/csi-driver-shared-resource/must-gather/gather

ENTRYPOINT /go/src/github.com/openshift/csi-driver-shared-resource/must-gather/gather
```
Resulted in this error message:

`[must-gather-c9hxz] POD 2022-01-25T16:08:10.164298602Z /bin/bash: /usr/bin/gather: No such file or directory`

```
FROM registry.ci.openshift.org/ocp/4.10:must-gather
WORKDIR /go/src/github.com/openshift/csi-driver-shared-resource
COPY . .
RUN chmod +x /go/src/github.com/openshift/csi-driver-shared-resource/must-gather/gather

ENTRYPOINT /go/src/github.com/openshift/csi-driver-shared-resource/must-gather/gather

```

did not produced any explicit error messages I could find, but the data in question was never collected.

Unless you have a Dockerfile different than one of those you think should work @adambkaplan that I can try, I believe we can claim due diligence here and accept the multi-stage build pattern I picked up from the other examples.

thanks